src/routes/docs.js

```diff
@@ -50,6 +61,42 @@ router.get('/:page', async (request, response, next) => {
 	response.render('docs/docs', renderData);
 });
 
+router.get([
+	'/errors/:errorCode', // TODO - Deprecate "errors" route
+	'/error/:errorCode',
+], async (request, response, next) => {
+	const locale = response.locals.localeString;
+	const error = request.params.errorCode;
+
+	const [ sysmodule, errorCode ] = error.split('-');
+
+	if (!errorCode) {
+		return next();
+	}
+
+	let template = fs.readFileSync(path.join(__dirname, '../../docs/common/error-page-template.md'), {
```

Comment:
I'd recommend doing something like:
```js
let template = await fs..promises(path.join(__dirname, '../../docs/common/error-page-template.md'), 'utf8');
```
So that you're not blocking the node thread each time it needs to respond to a request to this page.

Reply:
While this won't block the node thread, this increases the read time by quite a bit. Even the slowest instance of the sync version was faster than the fastest instance of the async version

Using the `promises` version of `readFile` in my testing takes between 0.331ms-1.141ms. Whereas the sync version takes between 0.059ms-0.168ms

Though at these speeds both are basically instant, so that may not matter much, but the sync version is so fast that the block is *also* essentially unnoticeable

Reply:
It's a tiny bit slower than sync, but it's being not time taken up on the node thread, so although it'll make each individual request a _tiny bit slower_ it means you can respond to more requests at once, either way it's probably not a massive difference. I'm just used to high traffic systems where we optimise for overall throughput above individual request times

Reply:
I would normally agree but even the response time was essentially the same. We use morgan to log response timings and in all my tests with both versions there was essentially no difference in response time

Also for context our site is relatively high traffic. In the past 7 days we've had 1.82m total requests (237.37k being in the past 24 hours) from 55.45k unique visitors. But we also use Cloudflare which handles a lot of caching for pages like this, so I'm not SUPER concerned about these kinds of micro optimizations right now

If it presents an issue later it can be addressed definitely, but as is there's no issues